core/src/program/blocks/call_block.rs

```diff
@@ -1,4 +1,12 @@
-use super::{fmt, hasher, Digest};
+use super::{fmt, hasher, Digest, Felt, Operation};
+
+// CONSTANTS
+// ================================================================================================
+
+/// Call domain used for hashing
+const CALL_DOMAIN: Felt = Felt::new(Operation::Call.op_code() as u64);
+/// Syscall domain used for hashing
+const SYSCALL_DOMAIN: Felt = Felt::new(Operation::SysCall.op_code() as u64);
```

Comment:
Instead of defining these as constants, maybe we could just add a `domain` method to `Operation` that returns the correct domain for all control blocks with a domain and zero otherwise. I think I would prefer this approach because it keeps it all encapsulated in one method that's attached to the operation type, rather than having the definitions spread across the constants in each code block file, especially since they're all doing the same thing. Wdyt?

Reply:
Yeah good call, I think that will be clearer!